mdtraj/formats/pdb/pdbstructure.py

```diff
@@ -49,8 +49,96 @@
 
 from mdtraj.core import element
 
-_residue_num_initial_nondecimal_vals = {"A000": "chimera", "2710": "hex", "****": "overflow"}
-_atom_num_initial_nondecimal_vals = {"A0000": "chimera", "186a0": "hex", "*****": "overflow"}
+
+def _overflow_residue_check(num_str, pdbstructure, curr_atom):
+    """
+    Function to check and guess what the current residue is because it's overflowed. Lifted from the 
+    original PDB code down below.
+
+    Parameters
+    ----------
+    num_str : str
+        str to be converted to int. Represents the residue number.
+    pdbstructure : PdbStructure
+        The PdbStructure object associated with `num_str`.
+    curr_atom : Atom
+        The current Atom object the residue number is associated with.
+
+    Returns
+    -------
+    int
+        The residue number deciphered in base 10.
+    """
+    if (
+        pdbstructure._current_model is None
+        or pdbstructure._current_model._current_chain is None
+        or pdbstructure._current_model._current_chain._current_residue is None
+    ):
+        # This is the first residue in the model.
+        return pdbstructure._next_residue_number
+    else:
+        currentRes = pdbstructure._current_model._current_chain._current_residue
+        if currentRes.name_with_spaces != curr_atom.residue_name_with_spaces:
+            # The residue name has changed.
+            return pdbstructure._next_residue_number
+        elif curr_atom.name_with_spaces in currentRes.atoms_by_name:
+            # There is already an atom with this name.
+            return pdbstructure._next_residue_number
+        else:
+            return currentRes.number
+
+
+_atom_num_initial_nondecimal_functions = {
+    "186a0": (lambda s: int(s, base=16)),  # "hex"
+    "A0000": (lambda s: (int(s[0], base=36) * 10**4 + int(s[1:], base=36))),  # "chimera"
+    "*****": (lambda s: self._next_atom_number),  # "overflow"
+}
+
+_residue_num_initial_nondecimal_functions = {
+    "2710": (lambda s, y=None, z=None: int(s, base=16)),  # 'hex'
```

Comment:
l noticed that all functions in the dictionary have three arguments, but do not use all of them. There are unused parameter `_num_str` in the `_overflow_residue_check` function and unused `y` and `x` in the hex and chimera parsers. From my read of its usage in the code `_read_residue_number`, there isn't a need to have a matching interface for these function, so the extra parameters could be removed. Overflow is only ever accessed explicitly with the "****" key for residue detection

If the functions were called more generically, this would make sense - could be some kind of future-proofing. For now, may be confusing.